src/lib/blkalloc/fixed_blk_allocator.cpp

```diff
@@ -89,11 +68,29 @@ BlkAllocStatus FixedBlkAllocator::alloc_contiguous(BlkId& out_blkid) { return al
 
 BlkAllocStatus FixedBlkAllocator::reserve_on_cache(BlkId const& b) {
     std::lock_guard lg(m_mark_blk_mtx);
-    HS_DBG_ASSERT(m_state == state_t::RECOVERING, "reserve_on_cache called on non-recovery path");
-    m_marked_blks.insert(b.blk_num());
+    if (m_state == state_t::RECOVERING) { m_marked_blks.insert(b.blk_num()); }
     return BlkAllocStatus::SUCCESS;
 }
 
+void FixedBlkAllocator::recovery_completed() {
```

Comment:
Moved the recovery processing from first alloc to the explicit call recovery_completed()